### DISCUSSION TITLE
Ability to pass files for linting to `next lint`

### DISCUSSION THREAD

**Discussion by jakeorr** [vercel/next.js#26564] [Ideas]
### Describe the feature you'd like to request

In a previous Next.js 10 project I had a package.json script for linting like this:

`"lint": "eslint --ext js,jsx,ts,tsx --fix",`

I was using [lint-staged](https://github.com/okonet/lint-staged)+husky to trigger a lint on pre-commit.

`lint-staged.config.js` ran the linting like this:

```
module.exports = {
  // Run ESLint on changes to JavaScript/TypeScript files
  '**/*.(ts|js)?(x)': filenames => `yarn lint ${filenames.join(' ')}`,
};
```

If I were to edit .eslintrc.js, `git add .` then attempt to commit, it would run a command like this, only linting files that are staged for commit:

`$ eslint --ext js,jsx,ts,tsx --fix .eslintrc.js`

On a new Next.js 11 project, I'm trying to use the same setup, but with `"lint": "next lint --fix",` as my package.json script. When lint-staged tries to run `$ next lint --fix .eslintrc.js`, it fails with:

```
> No `pages` directory found. Did you mean to run `next` in the parent (`../`) directory?
```

You should be able to see this by just running `npx next lint [filename]`.

I would like to be able to specify files that should be linted with `next lint`.

### Describe the solution you'd like

I would like to be able to pass file names to `next lint` and have it only lint those files.

`npx next lint [filename1] [filename2] [etc]`

### Describe alternatives you've considered

I would be fine with using a more custom linting solution, as long as I still get the benefits of `next lint` conformance. Do I need to run through `next lint` to get those? Would I be missing functionality by using the suggested eslint config, but just running `eslint`?

**Comment by housseindjirdeh** [comment]
Thanks for the detailed explanation of the issue you're facing!

I just opened #26777 which will let you pass in any file patterns like you normally would with the ESLint CLI.

> Do I need to run through next lint to get those? Would I be missing functionality by using the suggested eslint config, but just running eslint?

That's a great question. `next lint` does not need to be run in order to benefit from conformance rules. You just need to make sure you're extending from `eslint-config-next` or `eslint-plugin-next` within your `.eslintrc` file. The benefit of `next lint` is mostly from a developer experience standpoint, and more work is being done to [simplify the process](https://github.com/vercel/next.js/pull/26584) of setting up ESLint for first timers.

**tl.dr**: You can continue to run `eslint` as you currently are and still get the benefits of conformance, but the PR above also includes that functionality directly to `next lint` :)

**Reply by jakeorr** [reply]
This is a great answer, thank you! The details about using the new stuff with an existing eslint setup will work perfectly for me. I've been loving Next so far, and I really appreciate the support. 🥳

**Reply by housseindjirdeh** [reply]
Cheers! Thanks for all the feedback and if you notice anything else that can be improved, please let us know :)

**Reply by cryptiklemur** [reply]
Looks like this moved to a different PR: https://github.com/vercel/next.js/pull/28042
